codice/ddf
# DDF-4224 Make PreFederation and PostQuery filtering consistent

#### What does this PR do?
Merged the ValidationFilterPlugin (PreFederation) into the MetacardValidityFilterPlugin (PostQuery) so that their filtering configurations can remain in-sync to prevent unwanted or confusing behavior.

#### Who is reviewing it?
@pklinef @kcover @pvargas @ethantmanns @lamhuy @mdang8 
 
#### Select relevant component teams: 
@codice/io 

#### Ask 2 committers to review/merge the PR and tag them here.
@brendan-hofmann
@clockard

#### How should this be tested?
- On one machine, verify filter errors and filter warnings works with users that don't have the configured role.
- Verify user with configured role always receives metacards with errors/warning regardless of filterErrors/warnings config.
- Setup 2 machines, A and B.
Federate A to B. Configure B to not filter errors and warnings. Verify warning/error metacards are still federated to A.

#### Any background context you want to provide?

#### What are the relevant tickets?
[DDF-4224](https://codice.atlassian.net/browse/DDF-4224)

#### Screenshots

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
